src/mds/MDSContext.cc

```diff
@@ -119,7 +119,7 @@ void MDSLogContextBase::complete(int r) {
   MDLog *mdlog = get_mds()->mdlog;
   uint64_t safe_pos = write_pos;
   pre_finish(r);
-  // MDSContextBase::complete() free this
+  // MDSIOContextBase::complete() free this
   MDSIOContextBase::complete(r);
   mdlog->set_safe_pos(safe_pos);
```

Comment:
```suggestion
  // MDSIOContext::complete() free this
  mdlog->set_safe_pos(safe_pos);
  return MDSIOContextBase::complete(r);
```

Looks like you even found a small bug. The method should exit immediately (or return) if the object will be deleted.

Reply:
It seems that changes will lead to inconsistency between codes and comments in the header file. (Maybe I misunderstand the codes execution order and welcome to point out).
> MDSContext.h:120 Update log's safe pos after finish function return.

Reply:
Ah, well spotted. I guess leave the code as-is even though it's doing evil things. Please add a comment too that the safe_pos must be updated after `MDSIOContext::complete()` call.